[Woo POS][Refunds] Create Refund dialog

samiuelson opened the pull request:
## Summary
Implements the "Select items to refund" dialog UI for WooCommerce POS orders.

## What's New
- New refund dialog allowing users to select line items for refund
- Displays order items with product images, names, amounts, and tax columns
- "Select all" functionality for quickly selecting all items
- Continue button to proceed with the refund flow

samiuelson committed:
Implement 'Select items to refund' dialog UI in WooPos

- Update `WooPosIssueRefundDialog` to replace placeholder content with a list of selectable line items, displaying images, quantities, prices, and totals.
- Add `itemsSelectedLabel` to `WooPosOrdersState` and update `WooPosOrdersViewModel` to calculate and pass the selection count.
- Integrate the updated dialog into `WooPosOrdersScreen` passing the order line items.
- Add string resources for the refund selection screen headers and labels.

samiuelson committed:
Add padding for status and navigation bars to WooPosDialogWrapper

dangermattic commented:
<!--
  0 Errors
  1 Warning: This PR is larger than 300 lin...
  1 Message: This PR is still a Draft: some...
  0 Markdowns
-->
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is larger than 300 lines of changes. Please consider splitting it into smaller PRs for easier and faster reviews.</td>
    </tr>
  </tbody>
</table>
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          1 Message
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">This PR is still a Draft: some checks will be skipped.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

samiuelson committed:
Remove quantity and unit price display from WooPos order line items

- Remove `qtyAndUnitPrice` property from `LineItemRow` in `WooPosOrdersState`.
- Remove unit price calculation logic from `WooPosOrdersViewModel`.
- Remove UI elements displaying quantity and unit price in `WooPosIssueRefundDialog` and `WooPosOrdersDetails`.

samiuelson committed:
Improv Refund Dialog layout

- Add content description to the "Select All" checkbox for better accessibility.
- Add content description to individual line item checkboxes using the item name.
- Refactor hardcoded column width into a `COLUMN_WIDTH` constant.
- Switch `LazyColumn` usage from `items` to `itemsIndexed` for safer divider logic.

samiuelson committed:
Add test for correctly formatting the items selected label in the issue refund dialog

- Update `WooPosOrdersViewModelTest` to include a mock for `woopos_orders_items_selected` string resource.
- Add a new test case `given order with multiple line items, when onIssueRefundButtonClicked called, then itemsSelectedLabel is correctly formatted` to verify the label formatting.
- Verify the `resourceProvider` interaction in the existing `onIssueRefundButtonClicked` test.
